### ISSUE TITLE
brave_vpn should not use sources.gni

### ISSUE THREAD

**Issue by bridiver** [brave/brave-browser#27131]
sources.gni should only be used when there are circular dependencies

**Comment by spylogsster** [comment]
Need also remove VPN build and feature flag from `chromium_src/chrome/browser/net/stub_resolver_config_reader.cc` added in https://github.com/brave/brave-core/pull/15989 after this issue

**Comment by bsclifton** [comment]
@bridiver is this specifically referring to https://github.com/brave/brave-core/blob/master/browser/brave_vpn/sources.gni?

**Comment by bridiver** [comment]
Yes, the use of sources.gni should be limited to things that would otherwise create circular dependencies. If there are some classes that have circular dependencies that cannot easily be resolved then those classes should be in a separate directory because we should also not have BUILD.gn and sources.gni in the same directory
